<?xml version="1.0"?>
<launch>

    <node pkg="cob_robot_calibration_est" type="multi_step_cov_estimator.py" name="cal_cov_estimator_v0" output="screen" launch-prefix="time" args="/tmp/cal/cal_measurements.bag /tmp/cal" respawn="False"/>

    <group ns="calibration_config" clear_params="true">
      <!-- sensors -->
      <group ns="sensors">
          <param name="all" textfile="$(find cob_calibration_config)/$(env ROBOT)/user_defined/sensors.yaml" />
      </group>
      
      <param name="camera_parameter" value="$(find cob_calibration_data)/$(env ROBOT)/calibration/cameras/" />
 
      <!-- initial system for first calibration step -->
      <param name="initial_system" textfile="$(find cob_calibration_config)/$(env ROBOT)/autogenerated/system.yaml" />

      <!-- calibration steps -->
      <group ns="cal_steps">
        <group ns="step0">
          <param name="free_params" textfile="$(find cob_calibration_config)/$(env ROBOT)/modify/free_0.yaml" />
          <param name="use_cov" type="bool" value="True" />
          <rosparam>
            sensors:
            - arm_chain
            - kinect_rgb
          </rosparam>
          <param name="output_filename" type="string" value="result_step_1" />
        </group>
        <group ns="step1">
          <param name="free_params" textfile="$(find cob_calibration_config)/$(env ROBOT)/modify/free_1.yaml" />
          <param name="use_cov" type="bool" value="True" />
          <rosparam>
            sensors:
            - arm_chain
            - kinect_rgb
          </rosparam>
          <param name="output_filename" type="string" value="result_step_2" />
        </group>
        <group ns="step2">
          <param name="free_params" textfile="$(find cob_calibration_config)/$(env ROBOT)/modify/free_2.yaml" />
          <param name="use_cov" type="bool" value="True" />
          <rosparam>
            sensors:
            - arm_chain
            - kinect_rgb
          </rosparam>
          <param name="output_filename" type="string" value="result_step_3" />
        </group>
	<!--group ns="step3">
          <param name="free_params" textfile="$(find cob_calibration_config)/$(env ROBOT)/modify/free_3.yaml" />
          <param name="use_cov" type="bool" value="True" />
          <rosparam>
            sensors:
            - arm_chain
            - kinect_rgb
          </rosparam>
          <param name="output_filename" type="string" value="result_step_4" />
        </group>
	<group ns="step4">
          <param name="free_params" textfile="$(find cob_calibration_config)/$(env ROBOT)/modify/free_4.yaml" />
          <param name="use_cov" type="bool" value="True" />
          <rosparam>
            sensors:
            - arm_chain
            - kinect_rgb
          </rosparam>
          <param name="output_filename" type="string" value="result_step_5" />
        </group-->

      </group>
    </group>

</launch>
